rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryRepositoryImpl.scala

```diff
@@ -111,6 +112,38 @@ final case class HistoryRepositoryImpl[F[_]: Sync, C, P, A, K](
 
   type Result = (Blake2b256Hash, Option[PersistedData], HistoryAction)
 
+  protected[this] val dataLogger: Logger = Logger("coop.rchain.rspace.datametrics")
+
+  private def measure(actions: List[HotStoreAction]): Unit =
```

Comment:
This is a side-effecting function so it should be enclosed in the context `F` and it's execution delayed

Reply:
wrapped the logger in context F as suggested.
rchain.coop.shared.Log has a very nice logger implementation that we can use but it requires adding the 'share' project as a dependency.

Reply:
Isn't `shared` already a dependency of `rspace`? If not explicit then possibly transitive.
This sounds like something worth tackling in a separate PR

Reply:
yes